http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ResponseHandler.java

```diff
@@ -93,34 +102,65 @@ protected void channelRead0(ChannelHandlerContext channelContext, HttpObject msg
     }
 
     private static void finalizeRequest(RequestContext requestContext, ChannelHandlerContext channelContext) {
+
+        runAndLogError("Could not remove request specific handlers from pipeline",
+            () -> removePerRequestHandlers(channelContext));
         if (!channelContext.channel().attr(KEEP_ALIVE).get()) {
             closeAndRelease(channelContext);
         } else {
             requestContext.channelPool().release(channelContext.channel());
         }
+        channelContext.channel().attr(RESPONSE_COMPLETE_KEY).set(true);
     }
 
-    /**
-     * Close the channel and release it back into the pool.
-     *
-     * @param ctx Context for channel
-     */
-    private static void closeAndRelease(ChannelHandlerContext ctx) {
-        RequestContext requestContext = ctx.channel().attr(REQUEST_CONTEXT_KEY).get();
-        ctx.channel().close()
-           .addListener(channelFuture -> requestContext.channelPool().release(ctx.channel()));
+    private static void removePerRequestHandlers(ChannelHandlerContext ctx) {
+        removePerRequestHandlers(ctx, HttpStreamsClientHandler.class, ReadTimeoutHandler.class, WriteTimeoutHandler.class);
```

Comment:
I think we can get rid of this and remove them before the request starts.

Reply:
+1 looks like we were already doing that; removed